### ISSUE TITLE
Bug: this keyword scope is not working with version greater than 6.6.0

### ISSUE THREAD

**Issue by magarwal19** [vitalets/playwright-bdd#208]
**Given**
I create a step with say
When("I login to application",async function ({ context,  request}) {
this.loggedInUser = "test123";
}
**When**
I try to use the value stored with this keyword in last step to new step, then it is showing unidentified
e.g
Then("I validate logged in user",async function ({ context,  request}) {
console.log(this.loggedInUser)
}

**Then**
The value stored in first step is not accessible in next step. It is working on version 6.6.0

**But I expect**
we should be able to access value in next steps

**Environment**
platform: darwin
node: v16.20.0
playwright-bdd: v6.6.0
@playwright/test: v1.46.0
@cucumber/cucumber: v10.8.0
Playwright config file: playwright.config.js
```
```

**Comment by vitalets** [comment]
Hi @magarwal19 ! What steps style do you use?
From the provided code it's kind of mixed.
* If you use `{ context, request }` as a first argument of the step function, it is Playwright-style, so there is no `this` as a world. I've checked - you are right, it worked before, but it was never documented.
* if you use `this` in the step function, it is Cucumber-style -> arguments should contain only step parameters (empty in your example):
   ```js
   When("I login to application", async function () {
       this.loggedInUser = "test123";
   }
   ```

**Comment by magarwal19** [comment]
Hi @vitalets 
mostly we have playwright style steps where either we use context/browser/page/request

yes it was working before as all my tests are using this functionality and when updating to latest version it is breaking

**Comment by vitalets** [comment]
Got it. Then I'd suggest to move to either pure Playwright-style or pure Cucumber-style.

For pure **Playwright-style** you can define `world` fixture that holds `loggedInUser`:
```ts
type MyWorld = {
  loggedInUser: string;
};
export const test = base.extend<{ world: MyWorld }>({
  world: ({}, use) => use({ loggedInUser: '' }),
});
```
Then in steps use `world` instead of `this`:
```js
When("I login to application", async ({ context, request, world }) => {
   world.loggedInUser = "test123";
});

Then("I validate logged in user", async ({ context, request, world }) => {
   console.log(world.loggedInUser)
}):
```
For pure **Cucumber-style** you should also define `world` fixture, but it holds everything needed in steps: `loggedInUser`, `request`, `context` other fixtures:
```ts
type MyWorld = {
  loggedInUser: string;
  context: BrowserContext;
  request: ApiRequest;
};
export const test = base.extend<{ world: MyWorld }>({
  world: ({ request, context }, use) => use({ request, context, loggedInUser: '' }),
});

export const { Given, When, Then } = createBdd(test, { worldFixture: 'world' });
```
Then in steps you can access everything from `this`:
```js
When("I login to application", async function () {
   // this.context.newPage();
   this.loggedInUser = "test123";
});

Then("I validate logged in user", async function () {
   console.log(this.loggedInUser);
}):
```

**Comment by vitalets** [comment]
@magarwal19 could you try with just released **v7.2.2**?
I've added empty world `{}` to playwright-style steps.

**Comment by magarwal19** [comment]
Thank you for fixing this quickly!!

it is working now :)
